--- a/export-image/04-finalise/01-run.sh
+++ b/export-image/04-finalise/01-run.sh
@@ -72,11 +72,16 @@ unmount_image "${IMG_FILE}"
 
 mkdir -p "${DEPLOY_DIR}"
 
-rm -f "${DEPLOY_DIR}/image_${IMG_DATE}-${IMG_NAME}${IMG_SUFFIX}.zip"
+IMG_FILE_OLD="$IMG_FILE"
+IMG_FILE="${STAGE_WORK_DIR}/${IMG_NAME}-${IMG_DATE}${IMG_SUFFIX}.img"
+mv "$IMG_FILE_OLD" "$IMG_FILE"
+
+rm -f ${DEPLOY_DIR}/${IMG_NAME}-${IMG_DATE}${IMG_SUFFIX}.zip
+
+echo zip ${DEPLOY_DIR}/${IMG_NAME}-${IMG_DATE}${IMG_SUFFIX}.zip ${IMG_FILE}
 
 pushd "${STAGE_WORK_DIR}" > /dev/null
-zip "${DEPLOY_DIR}/image_${IMG_DATE}-${IMG_NAME}${IMG_SUFFIX}.zip" \
-	"$(basename "${IMG_FILE}")"
+zip ${DEPLOY_DIR}/${IMG_NAME}-${IMG_DATE}${IMG_SUFFIX}.zip $(basename ${IMG_FILE})
 popd > /dev/null
 
 cp "$INFO_FILE" "$DEPLOY_DIR"
